sqlglot/dialects/teradata.py

```diff
@@ -116,6 +118,16 @@ class Parser(parser.Parser):
             "TRANSLATE": lambda self: self._parse_translate(self.STRICT_CAST),
         }
 
+        EXPONENT = {
```

Comment:
now that we have to dialects that have this, can you refactor this out of postgres, and move it to the base parser?